homeassistant/components/shelly/config_flow.py

```diff
@@ -48,33 +52,68 @@ async def async_step_user(self, user_input=None):
         errors = {}
         if user_input is not None:
             try:
-                info = await self._async_get_info(user_input["host"])
+                self.host = user_input[CONF_HOST]
+                info = await self._async_get_info(self.host)
             except HTTP_CONNECT_ERRORS:
                 errors["base"] = "cannot_connect"
             except Exception:  # pylint: disable=broad-except
                 _LOGGER.exception("Unexpected exception")
                 errors["base"] = "unknown"
 
             else:
+                await self.async_set_unique_id(info["mac"])
```

Comment:
We need `self._abort_if_unique_id_configured()` here.
```suggestion
                await self.async_set_unique_id(info["mac"])
                self._abort_if_unique_id_configured()
```

Reply:
This did not have an abort before my changes as well, so I assume this was intentional.

Reply:
But without an abort the user could configure the device multiple times. I think missing abort was not intentional.

Reply:
We should abort indeed